src/parser/core/modules/CooldownDowntime.tsx

```diff
@@ -262,7 +263,11 @@ export abstract class CooldownDowntime extends Module {
 				gResets.shift()
 			}
 
-			while (charges < maxCharges && this.downtime.isDowntime(currentTime)) {
+			while (
+				currentTime < pullEndTimestamp
+				&& charges < maxCharges
+				&& this.downtime.isDowntime(currentTime)
+			) {
```

Comment:
added a check to this and the `if` below to prevent the `currentTime` trying to overrun the end of the pull

cc @AzariahAeroborn i think i followed the logic enough this should be safe, but more eyes would be nice.

Reply:
The later if probably doesn't need it, since the `while` loop this is all running within (start at 236) is already checking that condition, but I don't think it will hurt anything to add it to this inner while and might cover some oddities around end of fight downtime if the cast locks like e12s wind up triggering downtime detection.

Reply:
the `while` is incrementing `currentTime` - i'm imagining there's an instance where it'll get out of that while loop with `currentTime > pullEndTime`. if it does, `isDowntime`'ll throw.